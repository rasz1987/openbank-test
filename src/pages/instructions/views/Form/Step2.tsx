import { FC } from 'react';
import step from './step2.png';

const Step2: FC = () => {
  return (
    <>
      <img src={step} alt='step2' />
    </>
  )
};

export default Step2;