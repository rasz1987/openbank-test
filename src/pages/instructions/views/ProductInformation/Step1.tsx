import { FC } from 'react';
import step from './step1.png';

const Step1: FC = () => {
  return (
    <>
      <img src={step} alt="my alt step1" />
    </>
  )
};

export default Step1;


