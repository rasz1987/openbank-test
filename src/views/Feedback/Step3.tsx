import { FC } from 'react';
import { Step3Prop } from '../../interfaces';
import successImg from './success.png';
import errorImg from './error.png'

const Step3: FC<Step3Prop> = ({
  success = true
}) => {
  return <img src={success ? successImg : errorImg} alt="My alt" />
};

export default Step3;