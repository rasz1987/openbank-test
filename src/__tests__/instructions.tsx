import { render } from '@testing-library/react';
import Instrunctions from '../pages/instructions/Instructions';
import Step3 from '../pages/instructions/views/Feedback/Step3';
import Step2 from '../pages/instructions/views/Form/Step2';
import Step1 from '../pages/instructions/views/ProductInformation/Step1';

test('should render <Instrubctions /> without crash', async () => {
  const wrapper = render(<Instrunctions />);

  expect(wrapper).toMatchSnapshot();
});

test('should <Step3 /> without crash', async () => {
  const wrapper = render(<Step3 />);

  expect(wrapper).toMatchSnapshot();
});

test('should <Step2 /> without crash', async () => {
  const wrapper = render(<Step2 />);

  expect(wrapper).toMatchSnapshot();
});

test('should render <Step1 /> without crash', async () => {
  const wrapper = render(<Step1 />);

  expect(wrapper).toMatchSnapshot();
});