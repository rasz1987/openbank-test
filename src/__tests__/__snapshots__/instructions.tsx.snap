// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should <Step2 /> without crash 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="step2"
        src="step2.png"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="step2"
      src="step2.png"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should <Step3 /> without crash 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="My alt"
        src="success.png"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="My alt"
      src="success.png"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render <Instrubctions /> without crash 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <main
          class="App-content"
        >
          <h1>
            Bienvenid@ al test de 
            <img
              alt="openbank-logo"
              class="App-header-logo"
              src="key_openbank.png"
            />
          </h1>
          <h3>
            Objetivo 
          </h3>
          <p>
            Lo que pretendemos con la prueba es evaluar las capacidades técnicas respecto a un desarrollador web o front, especialmente en el area de React y aplicaciones SPA. Con esta prueba se pretende valorar muchos aspectos del stack tecnologico de un desarrollador del ambito web, como arquitectura, uso de patrones de diseño, maquetación, técnicas de programación, documentación, conocimentos de Javascript, HTML y CSS, entre otros. 
          </p>
          <h3>
            ¿En que consiste?
          </h3>
          <p>
            La prueba consiste en diseñar, maquetar y desarrollar desde cero un pequeño wizard que simule algo tan típico como la creación de una contraseña para un usuario. Lo que queremos que hagas es que plantees un wizard de unos 3 pasos en los que un usuario reciba información de que está a punto de crear una contraseña nueva, un formulario donde se le pidan una serie de datos para la creación de la contraseña y una página final de feedback de la operación.
          </p>
          <h3>
            Desarrollo
          </h3>
          <ul>
            <li>
              <h4>
                1º Pantalla de información de la contraseña
              </h4>
              <img
                alt="my alt step1"
                src="step1.png"
              />
              <p>
                En esta pantalla el usuario que entra a la aplicación debe entender que está contratando. Deberias darle la bienvenida al producto "Cuenta Corriente OpenClose" y explicarle que vas a demandarle sus datos en los siguientes pasos. Es una pantalla con una explicación, nombre del producto y botones de wizard de avanzar.
              </p>
              <h5>
                Esta pantalla será el paso numero uno del wizard y tiene que tener un check (que tiene que pulsar) que habilite el boton de avanzar, en el que se le pregunta al usuario si tiene mayoría de edad y que acepta que tratemos sus datos según la politica de protección de datos.
              </h5>
            </li>
            <li>
              <h4>
                2º Pantalla de creación de la contraseña
              </h4>
              <img
                alt="step2"
                src="step2.png"
              />
              <p>
                En esta pantalla al usuario se le van a pedir los siguientes datos en un formulario.
              </p>
              <ul>
                <li>
                  Contraseña(Min 8 - Max 24 caracteres)(Al menos 1 número y una mayúscula)
                </li>
                <li>
                  Pista (Maximo 255 caracteres)
                </li>
              </ul>
              <h5>
                Aquí para poder avanzar de paso, es necesario que todos los campos del formulario esten correctamente rellenos. Se debe controlar errores de tipo o forma en cada uno de ellos, mostrando un error asociado a cada input.  Se debe validar correctamente que sea una contraseña correcta y que coincidan ambas.
              </h5>
            </li>
            <li>
              <h4>
                3º Pantalla de feedback
              </h4>
              <img
                alt="My alt"
                src="success.png"
              />
              <img
                alt="My alt"
                src="success.png"
              />
              <p>
                Esta pantalla unicamente mostrará un icono de Ok/Ko informando al usuario del resultado, además de un boton que reinicie el proceso.
              </p>
              <p>
                Se provee de un mock en el services/api desde el cual se puede imitar el comportamiento de una llamada asíncrona. (Checkear para ver caso de error)
              </p>
              <h5>
                Debe ser una pantalla que cumpla cualquier de los requisitos, tanto un KO como un OK, de manera dinamica, de manera que, sea parametrizable en base al resultado de la creación.
              </h5>
            </li>
          </ul>
          <h3>
            Requisitos
          </h3>
          <p>
            Los requisitos que se van a tener en cuenta para la evaluación de la prueba son:
          </p>
          <ul>
            <li>
              Tiene que estar maquetada semánticamente correcta, con etiquetas de HTML5 y layouts actuales.
            </li>
            <li>
              Se puede renombrar, quitar o añadir cualquier recurso, fichero, asset que se crea necesario para mejorar o embellecer la prueba.
            </li>
            <li>
              La iconografia puede obtenerse de donde guste
            </li>
            <li>
              Nice to haves a tener en cuenta
              <ul>
                <li>
                  Estructura del proyecto
                </li>
                <li>
                  Control de la UI durante las llamadas asíncronas
                </li>
                <li>
                  Generalización de componentes y utilidades
                </li>
                <li>
                  Uso de buenas practicas de programación
                </li>
                <li>
                  Metodologias en hojas de estilos
                </li>
                <li>
                  Sistema de traducción
                </li>
                <li>
                  Rendimiento
                </li>
              </ul>
            </li>
          </ul>
          <h3>
            Recursos
          </h3>
          <p>
            Se provee un proyecto vacío donde se espera que el desarrollador complete la lógica. Ademas se provee de un base.scss con los colores corporativos e imágenes para completar los flujos. Todo esto es de libre disposición para que el destinatario lo use o no si lo cree conveniente.
          </p>
          <h3>
            Duración
          </h3>
          <p>
            No hay una estimación directa ni concreta de la prueba, pero queremos marcar como máximo 1 semana(7 días naturales) desde el recibo de la misma.
          </p>
          <h4
            class="bye-bye"
          >
            Sin más que añadir, ¡mucha suerte! y cualquier duda, mandanos tus dudas al correo que te han dado. =)
          </h4>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <main
        class="App-content"
      >
        <h1>
          Bienvenid@ al test de 
          <img
            alt="openbank-logo"
            class="App-header-logo"
            src="key_openbank.png"
          />
        </h1>
        <h3>
          Objetivo 
        </h3>
        <p>
          Lo que pretendemos con la prueba es evaluar las capacidades técnicas respecto a un desarrollador web o front, especialmente en el area de React y aplicaciones SPA. Con esta prueba se pretende valorar muchos aspectos del stack tecnologico de un desarrollador del ambito web, como arquitectura, uso de patrones de diseño, maquetación, técnicas de programación, documentación, conocimentos de Javascript, HTML y CSS, entre otros. 
        </p>
        <h3>
          ¿En que consiste?
        </h3>
        <p>
          La prueba consiste en diseñar, maquetar y desarrollar desde cero un pequeño wizard que simule algo tan típico como la creación de una contraseña para un usuario. Lo que queremos que hagas es que plantees un wizard de unos 3 pasos en los que un usuario reciba información de que está a punto de crear una contraseña nueva, un formulario donde se le pidan una serie de datos para la creación de la contraseña y una página final de feedback de la operación.
        </p>
        <h3>
          Desarrollo
        </h3>
        <ul>
          <li>
            <h4>
              1º Pantalla de información de la contraseña
            </h4>
            <img
              alt="my alt step1"
              src="step1.png"
            />
            <p>
              En esta pantalla el usuario que entra a la aplicación debe entender que está contratando. Deberias darle la bienvenida al producto "Cuenta Corriente OpenClose" y explicarle que vas a demandarle sus datos en los siguientes pasos. Es una pantalla con una explicación, nombre del producto y botones de wizard de avanzar.
            </p>
            <h5>
              Esta pantalla será el paso numero uno del wizard y tiene que tener un check (que tiene que pulsar) que habilite el boton de avanzar, en el que se le pregunta al usuario si tiene mayoría de edad y que acepta que tratemos sus datos según la politica de protección de datos.
            </h5>
          </li>
          <li>
            <h4>
              2º Pantalla de creación de la contraseña
            </h4>
            <img
              alt="step2"
              src="step2.png"
            />
            <p>
              En esta pantalla al usuario se le van a pedir los siguientes datos en un formulario.
            </p>
            <ul>
              <li>
                Contraseña(Min 8 - Max 24 caracteres)(Al menos 1 número y una mayúscula)
              </li>
              <li>
                Pista (Maximo 255 caracteres)
              </li>
            </ul>
            <h5>
              Aquí para poder avanzar de paso, es necesario que todos los campos del formulario esten correctamente rellenos. Se debe controlar errores de tipo o forma en cada uno de ellos, mostrando un error asociado a cada input.  Se debe validar correctamente que sea una contraseña correcta y que coincidan ambas.
            </h5>
          </li>
          <li>
            <h4>
              3º Pantalla de feedback
            </h4>
            <img
              alt="My alt"
              src="success.png"
            />
            <img
              alt="My alt"
              src="success.png"
            />
            <p>
              Esta pantalla unicamente mostrará un icono de Ok/Ko informando al usuario del resultado, además de un boton que reinicie el proceso.
            </p>
            <p>
              Se provee de un mock en el services/api desde el cual se puede imitar el comportamiento de una llamada asíncrona. (Checkear para ver caso de error)
            </p>
            <h5>
              Debe ser una pantalla que cumpla cualquier de los requisitos, tanto un KO como un OK, de manera dinamica, de manera que, sea parametrizable en base al resultado de la creación.
            </h5>
          </li>
        </ul>
        <h3>
          Requisitos
        </h3>
        <p>
          Los requisitos que se van a tener en cuenta para la evaluación de la prueba son:
        </p>
        <ul>
          <li>
            Tiene que estar maquetada semánticamente correcta, con etiquetas de HTML5 y layouts actuales.
          </li>
          <li>
            Se puede renombrar, quitar o añadir cualquier recurso, fichero, asset que se crea necesario para mejorar o embellecer la prueba.
          </li>
          <li>
            La iconografia puede obtenerse de donde guste
          </li>
          <li>
            Nice to haves a tener en cuenta
            <ul>
              <li>
                Estructura del proyecto
              </li>
              <li>
                Control de la UI durante las llamadas asíncronas
              </li>
              <li>
                Generalización de componentes y utilidades
              </li>
              <li>
                Uso de buenas practicas de programación
              </li>
              <li>
                Metodologias en hojas de estilos
              </li>
              <li>
                Sistema de traducción
              </li>
              <li>
                Rendimiento
              </li>
            </ul>
          </li>
        </ul>
        <h3>
          Recursos
        </h3>
        <p>
          Se provee un proyecto vacío donde se espera que el desarrollador complete la lógica. Ademas se provee de un base.scss con los colores corporativos e imágenes para completar los flujos. Todo esto es de libre disposición para que el destinatario lo use o no si lo cree conveniente.
        </p>
        <h3>
          Duración
        </h3>
        <p>
          No hay una estimación directa ni concreta de la prueba, pero queremos marcar como máximo 1 semana(7 días naturales) desde el recibo de la misma.
        </p>
        <h4
          class="bye-bye"
        >
          Sin más que añadir, ¡mucha suerte! y cualquier duda, mandanos tus dudas al correo que te han dado. =)
        </h4>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render <Step1 /> without crash 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="my alt step1"
        src="step1.png"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="my alt step1"
      src="step1.png"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
